### PR TITLE
Get the correct DB column name on filter/ sort by property

### PR DESCRIPTION
**Pull request by amitaibu**
#340

### CODE REVIEW

**Comment by amitaibu** [comment]
Next is tests

**Comment by amitaibu** [comment]
@mateu-aguilo-bosch ready for review.

File: plugins/restful/RestfulDataProviderEFQ.php [hunk 1]
```diff
@@ -151,16 +153,38 @@ protected function queryForListFilter(\EntityFieldQuery $query) {
...
+   * we make sure to get the real column name.
+   *
+   * @param string $property_name
+   *   The property name.
+   *
+   * @return string
+   *   The column name.
+   */
+  protected function getColumnFromProperty($property_name) {
+    $property_info = entity_get_property_info($this->getEntityType());
```
Comment (e0ipso) [inline plugins/restful/RestfulDataProviderEFQ.php:191]: Does this relate to the `'method'` in the public field definition? Should we leverage that?

File: tests/RestfulListTestCase.test [hunk 2]
```diff
@@ -259,28 +263,41 @@ class RestfulListTestCase extends RestfulCurlBaseTestCase {
       $roles = array_map('trim', explode(',', $item['roles']));
       $this->assertTrue(in_array($role_name, $roles), format_string('%role role found in content list.', array('%role' => $role_name)));
     }
+
+    // Sort by an entity metadata wrapper property that is different from the
+    // DB column name.
+    $handler = restful_get_restful_handler('articles', 1, 5);
+    $request = array('sort' => '-user',);
+    $result = $handler->get('', $request);
+    $this->assertEqual($result[0]['user']['id'], 3, 'List sorted by the "author" entity metadata wrapper property, which maps to the "uid" DB column name.');
```
Comment (e0ipso) [inline tests/RestfulListTestCase.test:272]: This test is relying on auto-created concrete user ids. We should only care about the ordering:

``` php
$this->assertEqual($result[0]['user']['id'], max($user1->uid, $user2->uid), 'List sorted by the "author" entity metadata wrapper property, which maps to the "uid" DB column name.');
```

File: tests/RestfulListTestCase.test [hunk 3]
```diff
@@ -373,6 +392,20 @@ class RestfulListTestCase extends RestfulCurlBaseTestCase {
...
+    $request = array(
+      'filter' => array(
+        'user' => array(
+          'value' => $user1->uid,
+        ),
+      ),
+    );
+
+    $result = $handler->get('', $request);
+    $this->assertEqual(count($result), 2, 'List filtered by the "author" entity metadata wrapper property, which maps to the "uid" DB column name.');
```
Comment (e0ipso) [inline tests/RestfulListTestCase.test:429]: ~~Same as above:~~

Comment (e0ipso) [inline tests/RestfulListTestCase.test:429]: Sorry, I mis-read the test. This is perfectly fine. :+1:

File: plugins/restful/RestfulDataProviderEFQ.php (hunk 1 above)
Comment (amitaibu) [inline plugins/restful/RestfulDataProviderEFQ.php:191]: > Does this relate to the 'method' in the public field definition?

No, we are just getting the real `schema field` from the entity metadata wrapper property definition.

**Comment by amitaibu** [comment]
I'll re-roll

**Comment by amitaibu** [comment]
Merged.

**Pull request closed without merging**